WordPress/src/main/java/org/wordpress/android/ui/posts/PostUploadService.java

```diff
@@ -894,20 +928,24 @@ public void updateNotificationWithError(String mErrorMessage, boolean isMediaErr
 
             String errorText = mContext.getResources().getText(R.string.upload_failed).toString();
             if (isMediaError) {
-                errorText = mContext.getResources().getText(R.string.media) + " " + mContext.getResources().getText(R.string.error);
-            }
-
-            mNotificationBuilder.setSmallIcon(android.R.drawable.stat_notify_error);
-            mNotificationBuilder.setContentTitle((isMediaError) ? errorText : mContext.getResources().getText(R.string.upload_failed));
-            mNotificationBuilder.setContentText((isMediaError) ? mErrorMessage : postOrPage + " " + errorText + ": " + mErrorMessage);
-            mNotificationBuilder.setContentIntent(pendingIntent);
-            mNotificationBuilder.setAutoCancel(true);
-
-            mNotificationManager.notify(mNotificationId, mNotificationBuilder.build());
+                errorText = mContext.getResources().getText(R.string.media) + " "
+                        + mContext.getResources().getText(R.string.error);
+            }
+
+            notificationBuilder.setSmallIcon(android.R.drawable.stat_notify_error);
+            notificationBuilder.setContentTitle((isMediaError) ? errorText :
+                    mContext.getResources().getText(R.string.upload_failed));
+            notificationBuilder.setContentText((isMediaError) ? mErrorMessage : postOrPage + " " + errorText
+                    + ": " + mErrorMessage);
+            notificationBuilder.setContentIntent(pendingIntent);
+            notificationBuilder.setAutoCancel(true);
+            mNotificationManager.notify(mNotificationId + (new Random()).nextInt(), notificationBuilder.build());
```

Comment:
If I understand this correctly, appending `Random().nextInt()` here means that every error will result in a new notification. If it's possible to have multiple errors for the same post upload then I don't think we want this behavior.